lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
```

Comment:
@jannyHou What does `statusCode` return on getting the current revision? Both on happy and unhappy cases.